app/domain/authentication/authn_gcp/extract_resource_restrictions.rb

```diff
@@ -0,0 +1,88 @@
+require 'command_class'
+
+module Authentication
+  module AuthnGCP
+
+    # This class is responsible of restrictions extraction that are set on a Conjur host or user as annotations.
+    ExtractResourceRestrictions = CommandClass.new(
+      dependencies: {
+        role_class:     ::Role,
+        resource_class: ::Resource,
+        logger:         Rails.logger
+      },
+      inputs:       %i(account username extraction_prefix)
+    ) do
+
+      def call
+        extract_resource_restrictions
+        restrictions_list
+      end
+
+      private
+
+      def extract_resource_restrictions
+        @logger.debug(LogMessages::Authentication::AuthnGCP::ExtractingRestrictionsFromResource.new(@username, @extraction_prefix))
+        prefixed_resource_annotations
+        init_restrictions_list
+        @logger.debug(LogMessages::Authentication::AuthnGCP::ExtractedResourceRestrictions.new(restrictions_list.length()))
+      end
+
+      def restrictions_list
```

Comment:
i think it will be better placed further down in the file. i like defining methods after we call them so it's easier to read through the file